### PR TITLE
T885 Add prepared statements support to sqlparser

### PR DESCRIPTION
**Pull request by storojs72**
1) Add `PREPARE stmt_name FROM preparable_stmt` statment support;
2) Add `EXECUTE stmt_name [USING @var_name [, @var_name] ...]` statement support;
3) Add `DEALLOCATE PREPARE stmt_name` statement support.

### CODE REVIEW

File: sqlparser/ast.go [hunk 1]
```diff
@@ -1338,137 +707,69 @@ type ShowFilter struct {
...
 
-func (node SelectExprs) walkSubtree(visit Visit) error {
-	for _, n := range node {
-		if err := Walk(visit, n); err != nil {
-			return err
-		}
-	}
-	return nil
+// DeallocatePrepare deallocates memory that stores compiled prepared statement
+type DeallocatePrepare struct {
```
[the remark this answers is not in the document]
Comment (storojs72) [reply sqlparser/ast.go]: `DeallocatePrepare` -> `DeallocatePreparedStatement` doesn't match to common names in ast. There are Select, Insert, Delete, Union - not SelectStatement, InsertStatement, DeleteStatement etc. `PreparedStatementName` -> `Name` - agree

File: tests/test.py [hunk 2]
```diff
@@ -1122,53 +1121,31 @@ def testBlacklist(self):
...
+                       "insert into test(id, data) values(1, DEFAULT)"]  # should be denied by pattern
 
+        for executor in executors:
+            for testQuery in testQueries:
+                with self.assertRaises(expectedException):
+                    executor.execute(testQuery)
+                try:
+                    executor.execute_prepared_statement(testQuery)
+                except psycopg2.ProgrammingError as e:
+                    self.assertTrue(str(e) == "no results to fetch")
```
[the remark this answers is not in the document]
Comment (storojs72) [reply tests/test.py]: I caught this error on insert in prepare statement. It seems that there is nothing to fetch when executing insert statement, but we try to do it inside function (https://github.com/cossacklabs/acra/blob/master/tests/test.py#L635)

**Comment by vixentael** [comment]
@Lagovas @storojs72 are there fixes to be made for this PR to be complete?

**Comment by storojs72** [comment]
Yes, one more - to handle case of ` PREPARE fooplan (int, text, bool, numeric) AS	INSERT INTO foo VALUES($1, $2, $3, $4); `

**Comment by storojs72** [comment]
@Lagovas look please this PR once more from the beginning

**Pull request merged**